examples/rh-sso/README.md

```diff
@@ -0,0 +1,112 @@
+
+This example shows you how to use Red Hat Single Sign-On to verify the identity of an [End-User](https://openid.net/specs/openid-connect-core-1_0.html#Terminology) and issue access tokens for use by clients wishing to access the End-User resources. 3scale will add an API Management layer, which includes access control, rate-limiting policies, and also a self-service model of registering the client applications for consuming the API.
+
+## Pre-requisites
+
+- 3scale account - You can sign up for a Free Trial [here](https://www.3scale.net/signup/)
+- Red Hat SSO instance - [Installation](https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.0/html-single/server_installation_and_configuration_guide/) and [Configuration](https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.0/html/getting_started_guide/) instructions
+- API - You can use the 3scale [echo API](https://echo-api.3scale.net:443)
+- Client Application to consume the API - You can use a client such as [Postman](https://www.getpostman.com/)
+- APIcast instance
+
+## Red Hat Single Sign-On Configuration for APIcast
+
+1. [Create a new realm](https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.0/html/getting_started_guide/create_a_realm_and_user#create-realm) (different from Master)
+2. Set up Tokens Policies (Realm Settings > Tokens) - e.g to configure access token ttl
+3. Set up [Initial Access Tokens](https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.0/html/securing_applications_and_services_guide/client_registration). This is necessary to synchronise client registrations between 3scale and Red Hat Single Sign-On.
+    - Define Expiration - This will determine how long the access token used to register clients from APIcas will be valid for, so make sure to choose a long lived value unless you want to be changing this token often.
+    - Define Count - This will determine how many clients can be registered using this access token, make sure to choose a large value unless you want to be changing this token often.
+4. [Create some test End-Users](https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.0/html/getting_started_guide/create_a_realm_and_user#create-new-user)
+
+## Client Registration in APIcast (Optional)
+
+In order to authenticate clients in RH SSO and correctly track usage in 3scale, client credentials need to be synchronised between the 2 systems. Typically this functionality would fall outside the realm of the Gateway, but we will show how you can implement this within the Gateway to provide a self contained environment.
+
+The following instructions are only required if you want to manage client registrations from APIcast. Otherwise you can skip this whole section.
+
+### Pre-requisites
+
+- Redis server needs to be running
+
+### APIcast Set up
+
+APIcast v2 should be installed as usual with the following additional dependencies:
```

Comment:
Writing stuff that changes to documentation does not make sense. It gets out of date and will be confusing. We can just write APIcast.